spec/challenge.spec.js

```diff
@@ -1,127 +1,104 @@
+  })
+
+  describe('enqueue()', () => {
+    it('adds aircrafts to the queue', () => {
+      aircrafts.forEach((aircraft, index) => {
+        this.atcQueue.enqueue(aircraft)
+
+        expect(this.atcQueue.aircraftCount()).to.equal(index + 1)
+      })
+    })
+  })
+
+  describe('dequeue()', () => {
+    it('removes aircrafts from the queue', () => {
+      aircrafts.forEach(aircraft => this.atcQueue.enqueue(aircraft))
+
+      for (let i = 1; i <= aircrafts.length; i++) {
+        this.atcQueue.dequeue()
+
+        expect(this.atcQueue.aircraftCount()).to.equal(aircrafts.length - i)
+      }
+    })
+
+    it('removes passenger aircrafts before cargo aircrafts', () => {
+      this.atcQueue.enqueue(cargoSmall)
+      this.atcQueue.enqueue(passengerSmall)
+      this.atcQueue.enqueue(passengerLarge)
+      this.atcQueue.enqueue(cargoLarge)
+
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'passenger')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'passenger')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'cargo')
+      expect(this.atcQueue.dequeue()).to.have.property('type', 'cargo')
+    })
+
+    it('removes large aircrafts before small aircrafts of the same type', () => {
+      this.atcQueue.enqueue(cargoSmall)
+      this.atcQueue.enqueue(passengerSmall)
+      this.atcQueue.enqueue(passengerLarge)
+      this.atcQueue.enqueue(cargoLarge)
+
+      expect(this.atcQueue.dequeue()).to.deep.equal(passengerLarge)
+      expect(this.atcQueue.dequeue()).to.deep.equal(passengerSmall)
+      expect(this.atcQueue.dequeue()).to.deep.equal(cargoLarge)
+      expect(this.atcQueue.dequeue()).to.deep.equal(cargoSmall)
+    })
+
+    it('removes earlier enqueued aircrafts of the same type and size first', () => {
+      const firstPassengerLarge = Object.assign({ order: 1 }, passengerLarge)
+      const secondPassengerLarge = Object.assign({ order: 2 }, passengerLarge)
+      const firstPassengerSmall = Object.assign({ order: 1 }, passengerSmall)
+      const secondPassengerSmall = Object.assign({ order: 2 }, passengerSmall)
+      const firstCargoLarge = Object.assign({ order: 1 }, cargoLarge)
+      const secondCargoLarge = Object.assign({ order: 2 }, cargoLarge)
+      const firstCargoSmall = Object.assign({ order: 1 }, cargoSmall)
+      const secondCargoSmall = Object.assign({ order: 2 }, cargoSmall)
+
+                                                  // Expected dequeue order
+      this.atcQueue.enqueue(firstCargoSmall)      // 7
+      this.atcQueue.enqueue(firstPassengerSmall)  // 3
+      this.atcQueue.enqueue(firstPassengerLarge)  // 1
+      this.atcQueue.enqueue(firstCargoLarge)      // 5
+      this.atcQueue.enqueue(secondCargoSmall)     // 8
+      this.atcQueue.enqueue(secondPassengerSmall) // 4
+      this.atcQueue.enqueue(secondPassengerLarge) // 2
+      this.atcQueue.enqueue(secondCargoLarge)     // 6
+
+      expect(this.atcQueue.dequeue()).to.deep.equal(firstPassengerLarge)
```

Comment:
Why did you move these all into one test? 
I think my goal in the original tests was to have them all be independent. Does it dequeue the right first plane, second plane, etc. I think having individual tests is better practice and better for the developers, but I could see the argument for doing it this way, so, I want to know why. :)